### ISSUE TITLE
Use env. variable in connections

### ISSUE THREAD

**Issue by klalafaryan** [airflow-helm/charts#183]
Is it possible to use env. variables from ConfigMap in the connections ? What would be the best solution for this ?

```
  connections:
    - id: s3_logs
      type: s3
      extra: |
        {
          "aws_access_key_id": "$AWS_S3_LOGS_ACCESS_KEY_ID",
          "aws_secret_access_key": "$AWS_S3_LOGS_SECRET_ACCESS_KEY"
        }
```

**Comment by hmahmood-slalom** [comment]
+1

**Comment by thesuperzapper** [comment]
@klalafaryan @hmahmood-slalom some questions:

1. what are you trying to achieve with this feature?
2. where would you be setting the environment variables like `AWS_S3_LOGS_ACCESS_KEY_ID`?

**Comment by hmahmood-slalom** [comment]
@thesuperzapper 

1. I'm trying to create Airflow connections where values for items such as password, etc. are using pre-existing Kubernetes secrets that I created ahead of time. Is there a way to do such thing?
2. I'm currently exposing the Kubernetes secrets as "extraEnv" variables.

**Comment by cfraleig** [comment]
@thesuperzapper 

As another workflow, we use ExternalSecrets to get values from AWS Secrets Manager and set them as k8s secrets. If we could add these as extraEnv and ref them in connections that would be great. Else the connections bit really isnt useful if we have to hard code keys into it. Consider just rotating keys and having this exposed or static.

**Comment by thesuperzapper** [comment]
@cfraleig @hmahmood-slalom  the `airflow.connections` value results in the creaton of the [./jobs/secret-job-scripts.yaml](https://github.com/airflow-helm/charts/blob/airflow-8.1.0/charts/airflow/templates/jobs/secret-job-scripts.yaml) Secret, which is mounted by the [./jobs/job-create-connections.yaml](https://github.com/airflow-helm/charts/blob/airflow-8.1.0/charts/airflow/templates/jobs/job-create-connections.yaml) Job, so its unclear where any environment variable you referenced would come from.

@hmahmood-slalom Are you asking for an extraEnv value associated with the Jobs, because [this is already the case](https://github.com/airflow-helm/charts/blob/airflow-8.1.0/charts/airflow/templates/jobs/job-create-connections.yaml#L41-L42) with `airflow.extraEnv` (but note those are mounted on all other pods also)?

**Comment by hmahmood-slalom** [comment]
@thesuperzapper I'm requesting the ability to be able to apply in the airflow.connections section the variables that I expose in the airflow.extraEnv section. Something along the lines of this:

`airflow:
      extraEnv:
            - name: SAMPLE_NAME
              valueFrom:
                   secretKeyRef:
                          name: "preexisting_secret"
                          key: "preexisting_key"
      connections:
             - ....
                password: (reference SAMPLE_NAME)`

We did discover a work-around where if you just create a Dag that runs a bash operator to run airflow cli commands to create the connections, you can reference the extraEnv variables in that command. 

Lastly, @thesuperzapper I'm not super familiar with Helm, but I did see in the documentation that the "lookup" function that comes with Helm might be useful, since it is essentially running kubectl command. I tried to use it in our template when creating the Airflow connections, but ran into syntax errors: https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-lookup-function

i.e.
      `connections:
             - ....
                password: (lookup "v1" "Secret" .Release.Namespace "preexisting_secret" ).data.password`

**Comment by thesuperzapper** [comment]
@klalafaryan @hmahmood-slalom  I have made a proposal to resolve this issue (I am working on implementing it now): https://github.com/airflow-helm/charts/issues/260

**Comment by thesuperzapper** [comment]
@klalafaryan @hmahmood-slalom and others watching, a feature similar to this has been implemented in [version 8.4.0](https://github.com/airflow-helm/charts/blob/main/charts/airflow/CHANGELOG.md#840---2021-07-09)!

**Comment by thesuperzapper** [comment]
@klalafaryan @hmahmood-slalom @cfraleig and others watching, a feature similar to this has been implemented in [version 8.4.0](https://github.com/airflow-helm/charts/blob/main/charts/airflow/CHANGELOG.md#840---2021-07-09)!
